### PR TITLE
Circle-matcher should allow create only one default circle by workspace

### PR DESCRIPTION
**Pull request by thallesfreitaszup**

### CODE REVIEW

File: circle-matcher/src/main/java/io/charlescd/circlematcher/domain/service/impl/SegmentationServiceImpl.java [hunk 1]
```diff
@@ -149,6 +152,26 @@ private boolean isItConvertibleToKv(Node node) {
...
     }
+
+    private void validateSegmentation(SegmentationRequest request) {
+        var isUpdate = request instanceof UpdateSegmentationRequest;
+
+        if (!request.getIsDefault()) {
+            return;
+        }
+        if (isUpdate) {
+            throw new IllegalArgumentException("Cannot update default segmentation");
```
Comment (leandrorodrigueszup) [inline circle-matcher/src/main/java/io/charlescd/circlematcher/domain/service/impl/SegmentationServiceImpl.java:163]: This is a domain Rule. I think we could use specific exceptions for that!

File: circle-matcher/src/main/java/io/charlescd/circlematcher/domain/service/impl/SegmentationServiceImpl.java [hunk 2]
```diff
@@ -149,6 +152,26 @@ private boolean isItConvertibleToKv(Node node) {
...
+        }
+
+        var metadataList = this.keyMetadataRepository.findByWorkspaceId(request.getWorkspaceId());
+        var metadataDefaultList = metadataList.stream().parallel()
+                .filter(metadata -> metadata.getIsDefault())
+                .collect(Collectors.toList());
+        if (!metadataDefaultList.isEmpty()) {
+            throw new IllegalStateException("Default segmentation already registered in workspace");
```
Comment (leandrorodrigueszup) [inline circle-matcher/src/main/java/io/charlescd/circlematcher/domain/service/impl/SegmentationServiceImpl.java:171]: This is a domain Rule. I think we could use specific exceptions for that!

File: circle-matcher/src/main/java/io/charlescd/circlematcher/domain/service/impl/SegmentationServiceImpl.java [hunk 3]
```diff
@@ -149,6 +152,26 @@ private boolean isItConvertibleToKv(Node node) {
...
+        if (!request.getIsDefault()) {
+            return;
+        }
+        if (isUpdate) {
+            throw new IllegalArgumentException("Cannot update default segmentation");
+        }
+
+        var metadataList = this.keyMetadataRepository.findByWorkspaceId(request.getWorkspaceId());
+        var metadataDefaultList = metadataList.stream().parallel()
+                .filter(metadata -> metadata.getIsDefault())
```
Comment (leandrorodrigueszup) [inline circle-matcher/src/main/java/io/charlescd/circlematcher/domain/service/impl/SegmentationServiceImpl.java:168]: Here is just a nitpicking :-D ... you could use method reference: `KeyMetadata::getIsDefault`

Comment (eliasalveszup) [reply circle-matcher/src/main/java/io/charlescd/circlematcher/domain/service/impl/SegmentationServiceImpl.java:168]: Or better... simpoly replace all this condition by `if (this.keyMetadataRepository.findByWorkspaceId(request.getWorkspaceId()).stream().anyMatch(KeyMetadata::getIsDefault))`

Comment (thallesfreitaszup) [reply circle-matcher/src/main/java/io/charlescd/circlematcher/domain/service/impl/SegmentationServiceImpl.java]: Done

**Review by eliasalveszup — APPROVED** [review]

**Review by leandrorodrigueszup — APPROVED** [review]

**Review by lucasbfernandes — APPROVED** [review]

**Pull request merged**